inc/classes/algolia/class-algolia-search.php

```diff
@@ -58,6 +68,9 @@ public function register_hooks(): void {
 
 		add_filter( 'get_the_terms', [ $this, 'get_post_terms' ], 10, 3 );
 		add_filter( 'wp_get_post_terms', [ $this, 'get_post_terms' ], 10, 3 );
+
+		// Block-theme compatibility: fix remote permalinks/excerpts in rendered blocks.
+		add_filter( 'render_block', [ $this, 'filter_render_block' ], 10, 3 );
```

Comment:
This was the one obvious thing from @danish17 's [diff](https://rtcamp.slack.com/archives/C08KC65RXFZ/p1764888820345009?thread_ts=1764877873.687419&cid=C08KC65RXFZ), figured I'd backport it now to get it out of the way before I migrate the class entirely and make the diff even worse.